data/campaigns/Eastern_Invasion/scenarios/04c_Mal-Ravanals_Capital.cfg

```diff
@@ -4,145 +4,421 @@
+    [event]
+        name=last breath #deliberate, so this triggers before ravanal_gets_serious
 
-    {LIMIT_CONTEMPORANEOUS_RECRUITS 2 "Skeleton Archer" 2}
-    {LIMIT_CONTEMPORANEOUS_RECRUITS 2 Ghoul 2}
+        [filter]
+            id=Jailer_{X}_{Y}
+        [/filter]
 
-    {LIMIT_CONTEMPORANEOUS_RECRUITS 3 Skeleton 2}
-    {LIMIT_CONTEMPORANEOUS_RECRUITS 3 "Vampire Bat" 2}
+        {MODIFY_UNIT id={ID} halo ""}
+        {MODIFY_UNIT id={ID} side 1}
+        [fire_event]
+            name=rescued_{ID}
+        [/fire_event]
+        [message]
+            speaker={ID}
+            message= _ "{MESSAGE}"
+        [/message]
+        [if]
+            [variable]
+                name=first_prisoner_freed
+                not_equals=yes
+            [/variable]
+            [then]
+                [message]
+                    speaker=Gweddry
+                    message= _ "How did these undead come to imprison you?"
+                [/message]
+                [message]
+                    speaker={ID}
+                    message= _ "It is a sad tale. My allies and I were questing when we were ambushed and captured by ghostly warriors. Their leader has been toying with us, forcing us to fight for their amusement."
+                [/message]
+                [message]
+                    speaker=Gweddry
+                    message= _ "No soldier deserves such treatment. We will try to free as many of you as possible."
+                [/message]
+                [message]
+                    speaker={ID}
+                    message= _ "Five of my comrades are still imprisoned. We will gladly enter your service in appreciation for our freedom."
+                [/message]
+            [/then]
+        [/if]
+        {VARIABLE first_prisoner_freed yes}
+        [if]
+            [not]
+                [have_unit]
+                    side=8
+                [/have_unit]
+            [/not]
 
-    {LIMIT_CONTEMPORANEOUS_RECRUITS 5 Shadow 1}
-    {LIMIT_CONTEMPORANEOUS_RECRUITS 5 Deathblade 3}
+            [then]
+                [set_achievement]
+                    content_for=eastern_invasion
+                    id=ei_S04c
+                [/set_achievement]
+            [/then]
+        [/if]
+    [/event]
+#enddef
+    [event] # this is a really ugly way to have an additional line for Terraent
+        name=rescued_Terraent # fired with rescued_{ID}
+        [message]
+            speaker=Gweddry
+            message=_ "A sun paladin! I had no idea there were any of your order still in Wesnoth."
+        [/message]
+    [/event]
+    [event]
+        name=prestart
+        {NAMED_NOTRAIT_UNIT 8 (Knight) 22 3 Aeraery (_"Aeraery")}
+        [+unit]
+            [modifications]
+                {TRAIT_SURVIVOR}
+                {TRAIT_QUICK}
+            [/modifications]
+        [/unit]
+        {IMPRISON Aeraery     21 3 (Thank you! My poor horse could barely breathe trapped among these reeds!)}
```

Comment:
I'm just reading the PR directly on GitHub, I haven't checked out the code, so I can't find where this `IMPRISON` macro comes from. Will it still allow the text to be translated?

Reply:
It needs the translation mark to be added. Unless Slayer95's proposed changes go in, wmlxgettext won't expand macros.